2016/proposals/deng-pan.md

```diff
@@ -0,0 +1,130 @@
+#Integrating `pandas.Panel` and xarray Features
+
+## Abstract
+
+Pandas package excels at processing tabular models, especially 1-D and 2-D models with data structure `pd.Series` and `pd.DataFrame`. However, despite the specialized data structure for 3-D and N-D data, `pd.Panel`, pandas is not well-designed for high dimensional data processing. Xarray was then developed to compensate the weak part of pandas. Xarray package enabled statistic and mathematical analysis of high-dimensional data, and implemented several new features that are missing in pandas.
+ 
+However, some useful features in pandas were not implemented in xarray yet. Thus, my project will be focusing on porting features from pandas to xarray, and improving data structure conversion between pandas and xarray. By the end of my work, I expect pandas/xarray users can have specialized tools for different types of data processing, and migrate between pandas and xarray smoothly. 
+
+## Technical Details
+
+Most of my proposal is supposed to be carried out with current implemented features in pandas and xarray. For PCA part, to improve the performance, I might switch to C++ and Eigen3 library.
+ 
+
+## Schedule of Deliverables
+
+###Milestone 1: Migrate features from pandas to xarray 
+
+**Week 1 - 5, May 23rd - June 26th**
+
+  - I will have my coursework finished within the first two weeks
+  - I will implement more multiIndex-related features and port features from pandas to xarray.
+
+    - **Week 1:** Enable multi-datatype in `xr.DataArray` like `pd.panel`
```

Comment:
xarray builds its core data structures on NumPy arrays, so this isn't feasible. It's also an intentional design decision not to build multiple dtypes into the array data structure -- we have `xr.Dataset` for multiple dtypes.